core/src/main/scala/kafka/api/FetchResponse.scala

```diff
@@ -112,6 +116,8 @@ case class TopicData(topic: String, partitionData: Map[Int, FetchResponsePartiti
 
 class TopicDataSend(val dest: String, val topicData: TopicData) extends Send {
 
+  private val emptyBuffer = ByteBuffer.allocate(0)
```

Comment:
We could do that. A few concerns: it's a bit strange that the empty buffer lives in `ErrorMapping`, it doesn't seem to be used anywhere at the moment and `ByteBuffer` has a number of mutation methods (although most of them don't do anything for an empty buffer). If you think it's worth having a shared empty buffer, maybe we could put it in `ApiUtils` as a private[api] value?